tests/CustomEdgeClassTest.php

```diff
@@ -0,0 +1,118 @@
+<?php
+/**
+ * ArangoDB PHP client testsuite
+ * File: Database.php
+ *
+ * @package ArangoDBClient
+ * @author  Frank Mayer
+ */
+
+namespace ArangoDBClient;
+
+/**
+ * Class DatabaseTest
+ * Basic Tests for the Database API implementation
+ *
+ * @property Connection $connection
+ *
+ * @package ArangoDBClient
+ */
+class CustomEdgeClassTest extends
+    \PHPUnit_Framework_TestCase
+{
+
+    protected static $testsTimestamp;
+
+    protected $collection;
+    protected $collectionHandler;
+
+    public function __construct($name = null, array $data = [], $dataName = '')
+    {
+        parent::__construct($name, $data, $dataName);
+        static::$testsTimestamp = str_replace('.', '_', (string) microtime(true));
+    }
+
+
+    public function setUp()
+    {
+        $this->connection        = getConnection();
+        $this->collectionHandler = new CollectionHandler($this->connection);
+
+        try {
+            $this->collectionHandler->drop('ArangoDB_PHP_TestSuite_TestCollection_01');
+        } catch (\Exception $e) {
+            // don't bother us, if it's already deleted.
+        }
+
+        $this->collection = new Collection();
+        $this->collection->setName('ArangoDB_PHP_TestSuite_TestCollection_01' . '_' . static::$testsTimestamp);
+        $this->collection->setType('edge');
+        $this->collectionHandler->create($this->collection);
+    }
+
+
+    /**
+     * Try to retrieve an edge with custom document class
+     */
+    public function testGetCustomEdgeWithHandler()
+    {
+        $connection      = $this->connection;
+        $collection      = $this->collection;
+        $edge            = new Edge();
+        $edgeHandler     = new EdgeHandler($connection);
+
+        $edge->someAttribute = 'someValue';
+        $edge->setFrom('test/v1');
+        $edge->setTo('test/v2');
+        $documentId = $edgeHandler->saveEdge($collection->getName(), 'test/v1', 'test/v2', $edge);
```

Comment:
However param 2 and 3 looks unnecessary since you can set it from edge.
How about remove it and change it like this

`saveEdge($collectionName, Edge $edge)`